session/session_test.go

```diff
@@ -445,51 +464,53 @@ func TestSessionBindRunsOnSessionBind(t *testing.T) {
 	err := errors.New("some error occured")
 	tables := []struct {
 		name          string
-		onSessionBind func(ctx context.Context, s *Session) error
+		onSessionBind func(ctx context.Context, s Session) error
 		err           error
 	}{
-		{"successful_on_session_bind", func(ctx context.Context, s *Session) error {
-			affectedVar = s.uid
+		{"successful_on_session_bind", func(ctx context.Context, s Session) error {
+			affectedVar = s.UID()
 			return nil
 		}, nil},
-		{"failed_on_session_bind", func(ctx context.Context, s *Session) error { return err }, err},
+		{"failed_on_session_bind", func(ctx context.Context, s Session) error { return err }, err},
 	}
 
 	for _, table := range tables {
 		t.Run(table.name, func(t *testing.T) {
 			affectedVar = ""
-			ss := New(nil, true)
+			sessionPool := NewSessionPool().(*sessionPoolImpl)
+			ss := sessionPool.NewSession(nil, true)
 			assert.NotNil(t, ss)
 
-			OnSessionBind(table.onSessionBind)
-			defer func() { sessionBindCallbacks = make([]func(ctx context.Context, s *Session) error, 0) }()
+			sessionPool.OnSessionBind(table.onSessionBind)
+			defer func() { sessionPool.sessionBindCallbacks = make([]func(ctx context.Context, s Session) error, 0) }()
 
 			uid := uuid.New().String()
 			err := ss.Bind(nil, uid)
 
 			if table.err != nil {
 				assert.Equal(t, table.err, err)
 				assert.Empty(t, affectedVar)
-				assert.Empty(t, ss.uid)
+				assert.Empty(t, ss.UID())
 			} else {
 				assert.NoError(t, err)
 				assert.Equal(t, uid, affectedVar)
-				assert.Equal(t, uid, ss.uid)
+				assert.Equal(t, uid, ss.UID())
 			}
 		})
 	}
 }
 
 func TestSessionBindFrontend(t *testing.T) {
-	ss := New(nil, true)
+	sessionPool := NewSessionPool().(*sessionPoolImpl)
+	ss := sessionPool.NewSession(nil, true)
 	assert.NotNil(t, ss)
 
 	uid := uuid.New().String()
 	err := ss.Bind(nil, uid)
 	assert.NoError(t, err)
-	assert.Equal(t, uid, ss.uid)
+	assert.Equal(t, uid, ss.UID())
```

Comment:
These interfaces are a little bit weird. They expose all of the things that the old struct did, so the only reason for them to exist is for testability purposes, which is not ideal.